Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Slate/FollowMeToggle.cs

```diff
@@ -1,40 +1,183 @@
+        /// </summary>
+        public Transform AutoFollowTransform
+        {
+            get { return autoFollowTransform; }
+            set
+            {
+                autoFollowTransform = value;
+
+                if (autoFollowTransform == null)
+                {
+                    autoFollowTransform = transform;
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("Optional transform to use when using autoFollowAtDistance. If not specified the local transform is used.")]
+        private Transform autoFollowTransform = null;
+
         private RadialView radialView = null;
+        private Coroutine autoFollowDistanceCheck = null;
 
-        private void Start()
+        #region MonoBehaviour Implementation
+
+        private void Awake()
         {
-            // Get Radial Solver component
             radialView = GetComponent<RadialView>();
+
+            if (autoFollowTransform == null)
+            {
+                autoFollowTransform = transform;
+            }
+
+            if (autoFollowAtDistance)
+            {
+                autoFollowDistanceCheck = StartCoroutine(AutoFollowDistanceCheck());
+            }
         }
 
+        #endregion MonoBehaviour Implementation
+
+        /// <summary>
+        /// Toggles the current follow behavior of the solver.
+        /// </summary>
         public void ToggleFollowMeBehavior()
         {
             if (radialView != null)
+            {
+                SetFollowMeBehavior(!radialView.enabled);
+            }
+        }
+
+        /// <summary>
+        /// Enables or disables the solver based on the follow parameter.
+        /// </summary>
+        /// <param name="follow">True if the solver should be active.</param>
+        public void SetFollowMeBehavior(bool follow)
+        {
+            if (radialView != null && radialView.enabled != follow)
             {
                 // Toggle Radial Solver component
                 // You can tweak the detailed positioning behavior such as offset, lerping time, orientation type in the Inspector panel
-                radialView.enabled = !radialView.enabled;
+                radialView.enabled = follow;
 
-                if(visualizationObject != null)
+                if (visualizationObject != null)
                 {
-                    visualizationObject.SetActive(radialView.enabled);
+                    visualizationObject.SetActive(follow);
+                }
+
+                if (interactableObject != null)
+                {
+                    interactableObject.SetDimensionIndex(follow ? 0 : 1);
                 }
             }
+        }
+
+        /// <summary>
+        /// Coroutine which checks how far away this transom is from the user and enables the follow behavior at a specified distance.
+        /// </summary>
+        /// <returns>Coroutine enumerator.</returns>
+        private IEnumerator AutoFollowDistanceCheck()
```

Comment:
You are calling StartCoroutine several times without ever stopping the coroutine. This will cause many coroutines to spin up, when I think you only ever want one to be running at a time. Make sure to stop any existing ones before running the coroutine.

Reply:
I think we are okay here due to the above `if (autoFollowAtDistance != value)` conditional safeguarding that case.